### PR TITLE
Docs: Display emoji/notice for rules with suggestions

### LINKED ISSUE #859: Docs: Add emoji and notice to indicate which rules provide suggestions
A number of core ESLint rules can provide automated suggestions. I would like to update the ESLint website (rules list and individual rule docs) to indicate which rules provide these suggestions.

* Add a 💡 emoji (represents idea/suggestion) next to rules that provide suggestions in the rules list/table
* Add the following to the top of individual rule docs:
  * 💡 Some problems reported by this rule are manually fixable by editor [suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).

Benefits of this change:

* Similar to how it's useful to know which rules have an autofixer, it's also useful to know which rules provide suggestions, as such functionality can enable easier adoption and usage of a rule
* Mentioning suggestions can help increase awareness of the relatively-new suggestions feature that was only released in [2019-11](https://eslint.org/blog/2019/11/eslint-v6.7.0-released) -- encouraging greater usage of it and the addition of suggestions to more rules

A number of plugins I work on have already added the suggestion emoji/notice to their rules table and individual rule docs:
* https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin#supported-rules
* https://github.com/platinumazure/eslint-plugin-qunit#rules
* https://github.com/sindresorhus/eslint-plugin-unicorn#rules
* https://github.com/ember-cli/eslint-plugin-ember#components
* https://github.com/ember-template-lint/ember-template-lint#rules

And in fact, enabling plugins to reliably show the above emoji/notice is one of the reasons why I pushed for requiring rules with suggestions to enable `meta.hasSuggestions` for us to key off of (https://github.com/eslint/eslint/pull/14573).

### PR DESCRIPTION
**Pull request by bmish**
Fixes #859.

Other part required for this change: https://github.com/eslint/eslint/pull/14830

### CODE REVIEW

**Comment by nzakas** [comment]
Force-pushing seems to be messing up the deploy preview. There’s no need to force-push, you can keep adding new commits. We only check the first commit message so the pressure is off. :)

can you try adding a new commit to see if it triggers the deploy preview to regenerate?

**Comment by bmish** [comment]
Will rules.yml get automatically regenerated at some point? I manually updated the relevant parts.

File: docs/rules/README.md
Comment (mdjermanovic) [inline docs/rules/README.md:7]: This is an old, unused file. If we want to update this text on [the rules page](https://deploy-preview-860--eslint.netlify.app/docs/rules/), we should update it in [docs/rules/index.liquid](https://github.com/eslint/website/blob/master/docs/rules/index.liquid#L8-L9)

**Comment by mdjermanovic** [comment]
> Will rules.yml get automatically regenerated at some point? I manually updated the relevant parts.

Yes, it gets updated on each ESLint release. (not including prereleases)

File: docs/rules/README.md
Comment (bmish) [reply docs/rules/README.md]: Good catch, updated new file.

**Review by nzakas — DISMISSED** [review]
LGTM. Thanks!

**Comment by mdjermanovic** [comment]
Preview LGTM!

Note about merging: If we want to merge this after we merge https://github.com/eslint/eslint/pull/14830 and do the release, then we should undo the changes in `_data/rules.yml`. Otherwise, it will produce an invalid yml with duplicate keys.

**Comment by bmish** [comment]
Sounds good. Just reverted `_data/rules.yml`. Mind re-approving / merging?

**Review by nzakas — APPROVED** [review]

**Pull request merged**
